src/main.js

```diff
@@ -434,15 +482,20 @@ function loadKnowledgeAppContent(app, knowledgeSearchProxy, uri, contentType, ti
     context.set_timestamp(timestamp);
 
     if (!knowledgeSearchProxy) {
-        app.launch([], context);
+        log('Without a known search proxy, app ' + app.get_id() + ' cannot be launched directly');
```

Comment:
`g_desktop_app_info_launch` won't work without a valid binary. Also, we probably don't want to do that from within the sandbox anyway.